Makefile

```diff
@@ -1,40 +1,71 @@
-.PHONY: install install-all dev dev-all tests lint docs clean build metadeps updatereqs
+.PHONY: .venv update-pip requirements install install-all dev dev-all tests lint docs doctest doctest2 clean build metadeps updatereqs
 
-install:
-	pip install .
+# Try to autodetect if python3 or python is the python executable used.
+PYTHON := $(shell which python3 2>/dev/null || which python 2>/dev/null)
+PIP := $(shell which pip3 2>/dev/null || which pip 2>/dev/null)
+PYTHONPATH=
+SHELL=/bin/bash
+VENV=.venv
 
-install-all:
-	pip install .[all]
+all:
+ifeq ($(PYTHON),)
+	$(error "No python interpreter found")
+endif
+	@echo Using $(PYTHON)
 
-dev:
-	pip install -e .
+.venv:  ## Set up a new Python virtual environment and install requirements
+	rm -rf .venv/
+	$(PYTHON) -m venv $(VENV)
+	$(MAKE) requirements
```

Comment:
Originally, I did have requirements use `$(VENV_BIN)/python` etc.) but then I didn't want to limit to just using the venv environment (I often use Anaconda); however, that is a good point that requirements will install to the base environment... Hmmm, need to either go all in venv or get rid of the requirements call.

Reply:
Yeah, I think since you provide full instructions at https://github.com/PECOS-packages/PECOS/tree/development#installation this `venv` target might be unnecessary (or as you write -- just get rid of requirements call here). It will be a one time setup for most developers (except the `source .venv/bin/activate` step that cannot be issued from `Makefile`).

Reply:
I understand, but I think it provides a false impression that it will **fully** setup the virtual environment. The user still has to issue `source .venv/bin/activate` to activate the env.

Reply:
An alternative might be to delete the last line of this target. Then the workflow will be to use `make venv`, switch to the environment and then issue `make requirements`.

Reply:
Hmmm, I don't feel like it is a big issue to build the venv with the requirements ready to go. From my understanding using things such as `$(VENV_BIN)/pip install` etc. in requirements will make the venv and not base environment update. However, it is easy enough to separate them as I am mainly using `venv` in other steps and can follow it with `requirements`. Given there might be confusion, I'll separate them.